### DISCUSSION TITLE
CPU/memory resource constraints are not enforced

### DISCUSSION THREAD

**Discussion by cristianrgreco** [aquasecurity/trivy#7343] [Bugs]
### Description

I've installed the trivy-operator via Helm. I've set the following resource constraints:

```yaml
resources:
  requests:
    cpu: 500m
    memory: 512Mi
  limits:
    cpu: 500m
    memory: 512Mi
```

My node has 4 total CPU cores. Observe that when the trivy-operator schedules the jobs, it seems these resource constraints are applied _per job_, resulting in almost 100% CPU utilisation. Because the operator creates a job per namespace, it's not possible to set the resource constraints to guarantee not exceeding a maximum value.

<img width="945" alt="image" src="https://github.com/user-attachments/assets/a58b1d95-ba1c-4164-811d-4291dc1109e4">

<img width="948" alt="image" src="https://github.com/user-attachments/assets/b97a29c0-2b6d-433c-a7a7-53e92f09565a">

### Desired Behavior

The trivy operator should not schedule more jobs than exceed the defined resource constraints.

### Actual Behavior

The trivy operator schedules all the jobs, potentially using 100% CPU/memory usage and grinding the cluster to a halt.

### Reproduction Steps

```bash
1. A kubernetes cluster with several workloads.
2. A fixed number of nodes in the cluster, e.g 1.
3. When Trivy runs it will likely utilise most cluster resources, regardless of resource constraints.
```

### Target

Kubernetes

### Scanner

Vulnerability and configuration

### Output Format

None

### Mode

Standalone

### Debug Output

```bash
N/A, running via operator
```

### Operating System

Ubuntu Server 24.04

### Version

```bash
Helm chart version 0.7.0
```

### Checklist

- [X] Run `trivy clean --all`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by simar7** [comment]
@afdesk could you take a look?
